### PR TITLE
Add filter docs

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This adds documentation of the collection's filters. For json_query, I copied the documenation from https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/user_guide/playbooks_filters.rst#selecting-json-data-json-queries.

I uploaded a docsite build to https://ansible.fontein.de/collections/community/general/ where you can see the results (https://ansible.fontein.de/collections/community/general/docsite/filter_guide.html).

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docsite

### CODE REVIEW

**Comment by felixfontein** [comment]
The sanity test failure is to be expected, since rstcheck doesn't know about the `ansible-output` lexer. I'll try out to modify antsibull-lint to accept it.

**Comment by felixfontein** [comment]
Failing CI will be fixed by https://github.com/ansible-community/antsibull/pull/278.

**Comment by felixfontein** [comment]
CC @aminvakil @russoz @Andersson007 @acozine @samccann @lmalivert

File: docs/docsite/rst/filter_guide.rst [hunk 1]
```diff
@@ -0,0 +1,753 @@
+.. _ansible_collections.community.general.docsite.filter_guide:
+
+Community.General Filter Guide
```
Comment (Akasurde) [inline docs/docsite/rst/filter_guide.rst:3]:
```suggestion
community.general Filter Guide
```

**Comment by russoz** [comment]
looking good to me

**Review by aminvakil — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@Akasurde @russoz @aminvakil thanks for your reviews!

**Pull request merged**